src/content/app/genome-browser/components/track-panel/components/track-panel-list/track-panel-items/TrackPanelGene.tsx

```diff
@@ -68,18 +67,55 @@ const TrackPanelGene = (props: TrackPanelGeneProps) => {
       trackId: GENE_TRACK_ID
     })
   );
-  const { toggleTrack, genomeBrowser } = useGenomeBrowser();
-
+  const browserGenomeTrackStates = useAppSelector(
+    getBrowserActiveGenomeTrackStates
+  );
+  const { toggleTrack, updateTranscriptTracks, genomeBrowser } =
+    useGenomeBrowser();
   const dispatch = useAppDispatch();
 
+  const transcriptTrackStatuses =
+    browserGenomeTrackStates?.transcriptTracks?.[focusObjectId] ?? {};
+  let sortedTranscripts: TrackPanelTranscriptType[] = [];
+
   useEffect(() => {
     toggleTrack({ trackId: GENE_TRACK_ID, status: trackStatus });
   }, [genomeBrowser]);
 
+  useEffect(() => {
+    if (sortedTranscripts.length) {
+      const visibleTranscriptIds = sortedTranscripts
+        .map((transcript) => transcript.stable_id)
+        .filter((transcriptId) => {
+          if (
+            transcriptTrackStatuses[transcriptId] === Status.SELECTED ||
+            transcriptTrackStatuses[transcriptId] === undefined
+          ) {
+            return true;
+          } else {
+            return false;
+          }
+        });
+
+      updateTranscriptTracks(visibleTranscriptIds);
+    }
+  }, [sortedTranscripts]);
```

Comment:
This gets called whenever an individual transcript track is toggled. Is there a better place to notify the genome browser on which transcripts need to be visible on load?

Reply:
You are re-declaring `sortedTranscripts` as a new array on line 79. I would expect this to mean that this useEffect will re-run every render cycle.

Reply:
Ah that's what I missed and was wondering why is it rerunning. Thanks for spotting this!